tests/unit_tests/data_steward/tools/participant_prevalence_test.py

```diff
@@ -0,0 +1,150 @@
+                               self.sandbox_dataset_id, self.pid_table_id)
+        self.assertIn(self.pid_project_id, actual)
+        self.assertIn(self.sandbox_dataset_id, actual)
+        self.assertIn(self.pid_table_id, actual)
+
+    @mock.patch('utils.bq.query')
+    def test_get_cdm_tables_with_person_id(self, mock_bq_query):
+        expected = self.cdm_pid_tables
+        columns_dict = {consts.TABLE_NAME_COLUMN: self.cdm_pid_tables}
+        mock_bq_query.return_value = pd.DataFrame(data=columns_dict)
+        actual = ptpr.get_cdm_tables_with_person_id(self.project_id,
+                                                    self.dataset_id)
+        self.assertEqual(actual, expected)
+
+    @mock.patch(
+        'cdr_cleaner.cleaning_rules.sandbox_and_remove_pids.get_tables_with_person_id'
+    )
+    @mock.patch('tools.participant_prevalence.get_cdm_tables_with_person_id')
+    def test_get_pid_counts_query(self, mock_pid_cdm_tables, mock_pid_tables):
+        mock_pid_cdm_tables.return_value = self.cdm_pid_tables
+        mock_pid_tables.return_value = self.pid_tables + self.cdm_pid_tables
+
+        actual = ptpr.get_pid_counts_query(self.project_id,
+                                           self.dataset_id,
+                                           self.hpo_id,
+                                           self.pids_string,
+                                           for_cdm=True)
+        self.assertEqual(len(self.cdm_pid_tables),
+                         len(actual.split(consts.UNION_ALL)))
+        for table in self.cdm_pid_tables:
+            self.assertIn(table, actual)
+
+        actual = ptpr.get_pid_counts_query(self.project_id,
+                                           self.dataset_id,
+                                           self.hpo_id,
+                                           self.pids_string,
+                                           for_cdm=False)
+        self.assertEqual(len(self.pid_tables),
+                         len(actual.split(consts.UNION_ALL)))
+        for table in self.pid_tables:
+            self.assertIn(table, actual)
+
+    @mock.patch('tools.participant_prevalence.get_pid_counts_query')
+    def test_get_pid_counts(self, mock_pids_query):
+        mock_pids_query.return_value = None
+        expected = pd.DataFrame(
+            columns=[consts.TABLE_ID, consts.ALL_COUNT, consts.EHR_COUNT])
+
+        actual = ptpr.get_pid_counts(self.project_id,
+                                     self.dataset_id,
+                                     self.hpo_id,
+                                     self.pids_string,
+                                     for_cdm=False)
+        pd.testing.assert_frame_equal(actual, expected)
+
+    @mock.patch('tools.participant_prevalence.get_pid_counts')
+    def test_get_non_zero_counts(self, mock_pid_counts):
+        cdm_counts_dict = {
+            consts.TABLE_ID: self.cdm_pid_tables,
+            consts.EHR_COUNT: self.zero_counts[:5],
```

Comment:
Does this mean the same thing (an array of 0's with the same length as `self.cdm_pid_tables`)?
```suggestion
            consts.EHR_COUNT: [0] * len(self.cdm_pid_tables),
```

Reply:
This comment might be obsoleted when `list[dict]` representation is used.